firestore/client.go

```diff
@@ -148,6 +148,12 @@ func (c *Client) Doc(path string) *DocumentRef {
 	return doc
 }
 
+// Query creates a query object related to this client.
+func (c *Client) Query() *Query {
```

Comment:
It looks like the semantics of the public API all are defined on a value type. Should the return a value too?

Reply:
Also it is not obvious to me how/when this method should be called. Could you provide an example?

Reply:
Should this be public?

Reply:
so, my thought here was to allow getting at Query().FromProto() for users. There may be a better way. This isn't strictly needed mind you, but I thought it made it a bit nicer.

Reply:
A thought: This isn't strictly necessary. 

```
client.Query().FromProto(runQueryRequest)

// vs

client.Collection("collectionId").Query.FromProto(runQueryRequest)
```


We could require the user to place the query into a collection first. I though this way was a bit more 'obvious' but it could also be added at later point